x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -335,135 +337,6 @@ private boolean containsWildcards(IndicesRequest indicesRequest) {
-                assert dateMathName.equals(aliasOrIndex) == false;
-                if (replaceWildcards && Regex.isSimpleMatchPattern(dateMathName)) {
-                    // continue
-                    aliasOrIndex = dateMathName;
-                } else if (authorizedIndices.contains(dateMathName) &&
-                    isIndexVisible(aliasOrIndex, dateMathName, indicesOptions, metadata, true)) {
-                    if (minus) {
-                        finalIndices.remove(dateMathName);
-                    } else {
-                        finalIndices.add(dateMathName);
-                    }
-                } else {
-                    if (indicesOptions.ignoreUnavailable() == false) {
-                        throw new IndexNotFoundException(dateMathName);
-                    }
-                }
-            }
-
-            if (replaceWildcards && Regex.isSimpleMatchPattern(aliasOrIndex)) {
-                wildcardSeen = true;
-                Set<String> resolvedIndices = new HashSet<>();
-                for (String authorizedIndex : authorizedIndices) {
-                    if (Regex.simpleMatch(aliasOrIndex, authorizedIndex) &&
-                        isIndexVisible(aliasOrIndex, authorizedIndex, indicesOptions, metadata)) {
-                        resolvedIndices.add(authorizedIndex);
-                    }
-                }
-                if (resolvedIndices.isEmpty()) {
-                    //es core honours allow_no_indices for each wildcard expression, we do the same here by throwing index not found.
-                    if (indicesOptions.allowNoIndices() == false) {
-                        throw new IndexNotFoundException(aliasOrIndex);
-                    }
-                } else {
-                    if (minus) {
-                        finalIndices.removeAll(resolvedIndices);
-                    } else {
-                        finalIndices.addAll(resolvedIndices);
-                    }
-                }
-            } else if (dateMathName == aliasOrIndex) {
-                // we can use == here to compare strings since the name expression resolver returns the same instance, but add an assert
-                // to ensure we catch this if it changes
-
-                assert dateMathName.equals(aliasOrIndex);
-                //Metadata#convertFromWildcards checks if the index exists here and throws IndexNotFoundException if not (based on
-                // ignore_unavailable). We only add/remove the index: if the index is missing or the current user is not authorized
-                // to access it either an AuthorizationException will be thrown later in AuthorizationService, or the index will be
-                // removed from the list, based on the ignore_unavailable option.
-                if (minus) {
-                    finalIndices.remove(aliasOrIndex);
-                } else {
-                    finalIndices.add(aliasOrIndex);
-                }
-            }
```

Comment:
Note that this clause was _not_ included in the code for the resolve index action because it would never apply to data streams. If you want to consolidate all that code (which would be great!), you'll need to incorporate that clause back into the common implementation.

Reply:
This _should_ be covered already on lines 107-113 on the common implementation, though it looks a little different because the comments have been removed and the equals logic is a bit more straightforward.